src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+//but till then the RoleAuthorizer has to be selected as the Authorizer Module of the proposal if the BountyManager is to be used
+
+contract BountyManager is IBountyManager, Module, PaymentClient {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using LinkedIdList for LinkedIdList.List;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    //@todo Reminder that this will be moved into the Module Contract at a later point of time
+    modifier onlyRole(uint8 roleId) {
+        if (
+            !IRoleAuthorizer(address(__Module_proposal.authorizer()))
+                .isAuthorized(roleId, _msgSender())
+        ) {
+            revert Module__BountyManager__OnlyRole(roleId, address(this));
+        }
+        _;
+    }
+
+    modifier onlyClaimContributor(uint claimId) {
+        address sender = _msgSender();
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+        uint length = contribs.length;
+        bool found;
+        for (uint i; i < length; i++) {
+            if (contribs[i].addr == sender) {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            revert Module__BountyManager__OnlyClaimContributor();
+        }
+        _;
+    }
+
+    modifier validPayoutAmounts(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount
+    ) {
+        if (
+            minimumPayoutAmount == 0
+                || maximumPayoutAmount < minimumPayoutAmount
+        ) {
+            revert Module__BountyManager__InvalidPayoutAmounts();
+        }
+        _;
+    }
+
+    modifier validBountyId(uint bountyId) {
+        if (!isExistingBountyId(bountyId)) {
+            revert Module__BountyManager__InvalidBountyId();
+        }
+        _;
+    }
+
+    modifier validClaimId(uint claimId) {
+        if (!isExistingClaimId(claimId)) {
+            revert Module__BountyManager__InvalidClaimId();
+        }
+        _;
+    }
+
+    function validContributorsForBounty(
+        Contributor[] memory contributors,
+        Bounty memory bounty
+    ) internal view {
+        //@update to be in correct range
+        uint length = contributors.length;
+        //length cant be zero
+        if (length == 0) {
+            revert Module__BountyManager__InvalidContributorsLength();
+        }
+        uint totalAmount;
+        uint currentAmount;
+        address contrib;
+        for (uint i; i < length; i++) {
+            currentAmount = contributors[i].claimAmount;
+
+            //amount cant be zero
+            if (currentAmount == 0) {
+                revert Module__BountyManager__InvalidContributorAmount();
+            }
+
+            contrib = contributors[i].addr;
+            if (
+                contrib == address(0) || contrib == address(this)
+                    || contrib == address(proposal())
```

Comment:
repeated function calls made to the `proposal()` function inside the loop. Cache it, perhaps?